src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -368,22 +408,76 @@
                 </if>
             </trim>
         </foreach>
+        </if>
+        )
     </sql>
 
+    <sql id="selectAllClinicalDataByAttribute">
+        SELECT sample_unique_id, patient_unique_id, attribute_value
+        FROM clinical_data_derived
+        WHERE attribute_name = '${clinicalDataFilter.attributeId}' AND type='${type}'
+        AND cancer_study_identifier IN
+        <foreach item="studyId" collection="studyViewFilterHelper.studyViewFilter.studyIds" open="(" separator="," close=")">
+            #{studyId}
+
+        </foreach>
+    </sql>
+    
     <sql id="categoricalClinicalDataCountFilter">
-        SELECT ${unique_id}
-        FROM ${table_name}
-        WHERE attribute_name = '${clinicalDataFilter.attributeId}' AND
-        type='${type}'
-        <foreach item="dataFilterValue" collection="clinicalDataFilter.values" open=" AND ((" separator=") OR (" close="))">
-            <trim prefix="" prefixOverrides="AND">
-                AND (
+        <!-- check if 'NA' is selected -->
+        <bind name="userSelectsNA" value="false" />
+        <bind name="userSelectsValue" value="false" />
+        <foreach item="dataFilterValue" collection="clinicalDataFilter.values">
+            <choose>
+                <when test="dataFilterValue.value == 'NA'">
+                    <bind name="userSelectsNA" value="true" />
+                </when>
+                <otherwise>
+                    <bind name="userSelectsValue" value="true" />
+                </otherwise>
+            </choose>
+        </foreach>
+        (
+        <!-- if 'NA' is selected, prepare NA samples -->
+        <if test="userSelectsNA">
+            SELECT DISTINCT ${unique_id}
+            FROM sample_derived sd
+            LEFT JOIN (<include refid="selectAllClinicalDataByAttribute"/>) AS categorical_clinical_data
+            ON
+            <choose>
+                <when test="'${type}' == 'sample'">
+                    sd.sample_unique_id = categorical_clinical_data.sample_unique_id
+                </when>
+                <otherwise>
+                    sd.patient_unique_id = categorical_clinical_data.patient_unique_id
+                </otherwise>
+            </choose>
+            WHERE empty(attribute_value)
```

Comment:
So for here you can bring in the WHERE conditions for non-NA samples like
```
WHERE
  for each dataFilterValue:
    if dataFilterValue is 'NA', empty(attribute_value)
    else, normalizedAttributeValue ILIKE datavalue.value
```
Then you can save a UNION